go-controller/pkg/ovn/master.go

```diff
@@ -1170,14 +1148,10 @@ func (oc *Controller) checkNodeChassisMismatch(node *kapi.Node) (bool, error) {
 	}
 
 	chassisList, err := libovsdbops.ListChassis(oc.sbClient)
-	if err != nil && err != libovsdbclient.ErrNotFound {
+	if err != nil {
 		return false, fmt.Errorf("failed to get chassis list for node %s: error: %v", node.Name, err)
 	}
 
-	if err == libovsdbclient.ErrNotFound {
```

Comment:
Per libovsdb client API, a `List*` operation does not fail with `ErrNotFound` when no entries are found, in that case it just returns an empty list. `ErrNotFound` is returned when the table itself does not exist in the cache which is more of an internal error for which unfortunately `ErrNotFound` was re-used. So in case of `List*` operations `ErrNotFound` should be treated as a legit error.